src/Main.elm

```diff
@@ -2,82 +2,172 @@ module Main exposing (..)
 
 import Browser
 import Html exposing (..)
-import Html.Events exposing (onClick)
 import Html.Attributes exposing (style)
+import Html.Events exposing (onClick)
+
+
 main =
-  Browser.sandbox { init = init, update = update, view = view }
+    Browser.sandbox { init = init, update = update, view = view }
+
+
+type Msg
+    = Click Int
+
+
+type Player
+    = Naughty
+    | Crossy
+
+
+type Square
+    = Blank
+    | Naught
+    | Cross
 
-type Msg = Click Int
-type Player = Naughty | Crossy
-type Square = Blank | Naught | Cross
 
-type alias Grid = List Square
+type alias Combination =
+    ( Square, Square, Square )
 
-type GameState = CurrentlyPlaying Player | Draw | Winner Player
-type alias Model = { grid: Grid, player: Player, gameState: GameState }
 
-init = { grid = List.repeat 9 Blank, player = Naughty, gameState = CurrentlyPlaying Naughty }
+type alias Grid =
+    List Square
 
-getSquare: Player -> Square
+
+type GameState
+    = CurrentlyPlaying Player
+    | Draw
+    | Winner Player
+
+
+type alias Model =
+    { grid : Grid, player : Player, gameState : GameState }
+
+
+init =
+    { grid = List.repeat 9 Blank, player = Naughty, gameState = CurrentlyPlaying Naughty }
+
+
+getSquare : Player -> Square
 getSquare player =
-  if (player == Naughty) then Naught else Cross
+    case player of
+        Naughty ->
+            Naught
+
+        Crossy ->
+            Cross
+
+f : Combination -> Maybe Player
+f combination = case combination of
```

Comment:
We should probably give this function a proper name now that we know what it does 😅